### PR TITLE
Fixed #24135 -- Made RenameModel rename many-to-many tables.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/24135

### CODE REVIEW

File: django/db/migrations/operations/models.py [hunk 1]
```diff
@@ -216,6 +216,26 @@ def database_forwards(self, app_label, schema_editor, from_state, to_state):
...
+            # Alter many-to-many fields affected by this model's name.
+            for field in old_model._meta.local_many_to_many:
+                # Skip self-referential fields as these are renamed above.
+                if field.model == field.related_model:
+                    continue
+                # Rename the M2M table that's based on this model's name.
+                old_m2m_model = old_model._meta.get_field(field.name).rel.through
+                new_m2m_model = new_model._meta.get_field(field.name).rel.through
+                schema_editor.alter_db_table(
+                    new_model,
```
Comment (charettes) [inline django/db/migrations/operations/models.py]: I think this should be `old_m2m_model`.

**Comment by charettes** [comment]
LGTM.

File: django/db/migrations/operations/models.py [hunk 2]
```diff
@@ -216,6 +216,27 @@ def database_forwards(self, app_label, schema_editor, from_state, to_state):
                     related_object.field,
                     to_field,
                 )
+            # Rename M2M fields whose name is based on this model's name.
+            fields = zip(old_model._meta.local_many_to_many, new_model._meta.local_many_to_many)
```
Comment (MarkusH) [inline django/db/migrations/operations/models.py]: Your assumption here is that the order of the fields doesn't change. I think we should check for the `to` model and the `through_fields`.

Comment (timgraham) [inline django/db/migrations/operations/models.py]: I copied that pattern from 41b337ef, could we table that improvement to a separate ticket?

Comment (MarkusH) [inline django/db/migrations/operations/models.py]: In that case, sure.

**Comment by MarkusH** [comment]
Looks good to me. If you want you could add some test data before the rename and check for correct value afterwards.

**Pull request merged**